gdrive_sync/api.py

```diff
@@ -557,18 +558,127 @@ def update_sync_status(website: Website, sync_datetime: datetime):
     website.save()
 
 
+def __iter_all_config_items(website: Website) -> Iterator[Tuple[bool, ConfigItem]]:
+    """
+    Yields ConfigItem for all active starters.
+
+    Args:
+        website (Website): The website being scanned.
+
+    Yields:
+        Iterator[Tuple[bool, ConfigItem]]: A generator where yield[0] indicates whether or not the ConfigItem
+            yield[1] belongs to the starter of `website`.
+    """
+    # Assume the starter of websites without a starter to be the course starter.
+    website_starter_slug = (
+        website.starter.slug if website.starter else settings.OCW_COURSE_STARTER_SLUG
+    )
+    all_starters = WebsiteStarter.objects.filter(
+        status__in=WebsiteStarterStatus.ALLOWED_STATUSES
+    )
+
+    for starter in all_starters:
+        for item in SiteConfig(starter.config).iter_items():
+            yield website_starter_slug == starter.slug, item
+
+
+def __iter_related_fields(
```

Comment:
I believe this could just be a function on `ConfigItem` directly called `iterate_fields` that takes an optional boolean argument of `only_cross_site`. We could also just put this in `utils.py`, but either way I tihnk `only_cross_site` should be optional and default to `False`.

Reply:
You're right, thanks. This function is generic enough to be a part of CofigItem.